matheusberger/pb-ios
# Create a navigation link to HomeView (pedal list view)

## Changes
- Refactored HomeView to receive a HomeViewModel
- Refactored HomeViewModel to receive a LocalDataProvider<Pedal>
- Created computed var to return a HomeViewModel in SongListViewModel
- Added a button in SongListView to navigate to HomeView
- Updated unit tests

### Screenshots
![Simulator Screenshot - iPhone 15 - 2023-12-28 at 18 54 27](https://github.com/matheusberger/pb-ios/assets/24720744/5d68b426-533a-4bb2-902f-afcea29ad95b)
![Simulator Screenshot - iPhone 15 - 2023-12-28 at 18 54 49](https://github.com/matheusberger/pb-ios/assets/24720744/6881ea42-d9ba-4876-aae2-a43059920a31)


## Affected Areas:
- SongListView and HomeView

## Test Procedure:
Run app, verify that the button does appear and that it correctly navigates to HomeView

## Unit Tests:
- Updated HomeViewTests to reflect that HomeView's init now receives a HomeViewModel
